src/arch/xtensa/include/xtensa/config/core.h

```diff
@@ -42,17 +42,22 @@
  * building SOF with Zephyr and XtensaTools.
  */
 
-#if defined(__ZEPHYR__)
+#if defined(_ASMLANGUAGE)
```

Comment:
```suggestion
#if !defined(_ASMLANGUAGE) && !defined(__ASSEMBLER__)
```
like in https://github.com/zephyrproject-rtos/zephyr/commit/f7538f0005fbbe70d92d1535656f4fc84e2d42a0

That Zephyr commit is from 2017 so it has stood the test of time.

Reply:
Done